packages/@aws-cdk/aws-servicecatalogappregistry/lib/attribute-group.ts

```diff
@@ -45,6 +53,22 @@ export interface AttributeGroupProps {
 abstract class AttributeGroupBase extends cdk.Resource implements IAttributeGroup {
   public abstract readonly attributeGroupArn: string;
   public abstract readonly attributeGroupId: string;
+
+  public shareResource(shareOptions: ShareOptions): void {
+    const principals = getPrincipalsforSharing(shareOptions);
+    const shareName = `RAMShare${this.generateUniqueHash(this.node.addr)}`;
+    new CfnResourceShare(this, shareName, {
+      name: shareName,
+      allowExternalPrincipals: shareOptions.allowExternalPrincipals ?? true,
+      principals: principals,
+      resourceArns: [this.attributeGroupArn],
+    });
+  }
+
+  /**
+   * Create a unique id
+   */
+  protected abstract generateUniqueHash(resourceAddress: string): string;
```

Comment:
We have methods in `Names` for this. Please use [`Names.uniqueResourceName`](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/core/lib/names.ts#L84) for this instead

Reply:
In my latest commit I gave a try at using this method.